drivers/storage/portworx/portworx.go

```diff
@@ -1287,6 +1286,46 @@ func setTLSDefaults(
 	}
 }
 
+func setTelemetryDefaults(
+	toUpdate *corev1.StorageCluster,
+	pxVersion *version.Version,
+) error {
+	// Disable telemetry if it's not supported but enabled explicitly
+	if pxutil.IsTelemetryEnabled(toUpdate.Spec) && pxVersion.LessThan(pxutil.MinimumPxVersionCCM) {
+		toUpdate.Spec.Monitoring.Telemetry.Enabled = false // telemetry not supported for < 2.8
+		toUpdate.Spec.Monitoring.Telemetry.Image = ""
+		toUpdate.Spec.Monitoring.Telemetry.LogUploaderImage = ""
+		return nil
+	}
+
+	// Only enable telemetry by default under conditions:
+	// 1. CCM Go is supported
+	// 2. Portworx initialized (cluster UUID ready)
+	// 3. telemetry is not specified explicitly
+	// 4. no air-gapped or custom proxy
+	if !pxutil.IsCCMGoSupported(pxVersion) ||
+		toUpdate.Status.ClusterUID == "" ||
```

Comment:
Is this check run repeatedly/more than just at startup?

Reply:
it is being checked repeatedly. operator only start telemetry reconcile once px is running, so something similar here.